### PR TITLE
Follow up of #2972 and tidy up left up rosettasciio

### PR DESCRIPTION
**Pull request by ericpre**
Follow up of #2972 to use https://github.com/hyperspy/rosettasciio.

### Progress of the PR
- [x] Remove rosettasciio folder,
- [x] update CI workflow accordingly, 
- [x] update dependency requirement, 
- [n/a] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [n/a] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [n/a] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: conda_environment.yml [hunk 1]
```diff
@@ -6,18 +6,17 @@ dependencies:
...
 - fsspec
-- h5py
-- imageio
+- importlib-metadata
 - ipyparallel
 - ipython !=8.0.*
 - jinja2
 - matplotlib-base
-- matplotlib-scalebar
 - mrcz
```
Comment (jlaehne) [inline conda_environment.yml]: Isn't this an IO dependency as well?

File: hyperspy/io.py [hunk 2]
```diff
@@ -446,7 +446,11 @@ def load(filenames=None,
         )
 
     if not filenames:
-        raise ValueError('No file(s) provided to reader.')
+        if pattern:
+            # in case, the file doesn't exist
+            raise ValueError(f'No filename matches the pattern "{pattern}"')
+        else:
+            raise ValueError('No file(s) provided to reader.')
```
Comment (jlaehne) [inline hyperspy/io.py]: Add coverage?

File: hyperspy/tests/test_io.py [hunk 3]
```diff
@@ -189,120 +188,110 @@ def test_file_not_found_error():
...
+            'operation': 'load',
+            'hyperspy_version': hs_version
+        },
+        '1': {
+            'io_plugin': 'rsciio.hspy.api',
+            'operation': 'save',
+            'hyperspy_version': hs_version
+        },
+        '2': {
+            'io_plugin': 'rsciio.hspy.api',
```
Comment (jlaehne) [inline hyperspy/tests/test_io.py:281]: This now duplicates the item `'0'` above. Is it just to make sure that the saved file can be reread?

Comment (ericpre) [reply hyperspy/tests/test_io.py:281]: They have a time difference, but in the test it is removed to simplify the test.

File: hyperspy/io.py (hunk 2 above)
Comment (ericpre) [reply hyperspy/io.py]: Actually, this branch is not necessary because this case is covered in lines 419-420 above.

**Review by ericpre — COMMENTED** [review]
Should be all done!

**Review by jlaehne — APPROVED** [review]

**Pull request merged**